xapian-core/tests/api_backend.cc

```diff
@@ -1401,3 +1401,35 @@ DEFINE_TESTCASE(embedded1, singlefile) {
 
     return true;
 }
+
+static string get_db_table_path(const string& db_path){
+    string db_table_path;
+    if(endswith(get_dbtype(), "glass")){
+        db_table_path = db_path + "/postlist.glass";
+    }else if(endswith(get_dbtype(), "chert")){
+        db_table_path = db_path + "/postlist.DB";
+    }
+    return db_table_path;
+}
+
+//check database table name. it could end with ".DB", or ".glass".
+DEFINE_TESTCASE(databasetablename, chert || glass){
+	//database not exist
+    const string & db_path_nonexist = get_named_writable_database_path("databasetablename");
+    string db_table_path = get_db_table_path(db_path_nonexist);
+    try{
+        Xapian::Database::check(db_table_path);
+    }catch(const Xapian::DatabaseOpeningError &){
+    }
+
```

Comment:
This will also pass if the exception is not thrown.  The `TEST_EXCEPTION` macro provides a standard way to check that an expected exception is definitely thrown.